R/deployApp.R

```diff
@@ -373,14 +386,15 @@ deploymentTarget <- function(appPath, appName, account, server = NULL) {
     }
     accountDetails <- accountInfo(account, server)
     createDeploymentTarget(
-      tools::file_path_sans_ext(basename(appPath)),
-      account, accountDetails$server)
+      generateAppName(appTitle, appPath, account),
```

Comment:
Presuming we're referring to the same conditional (https://github.com/rstudio/rsconnect/blob/691585f43aef947284789da50a95609eb9981922/R/deployApp.R#L331-L334), that name is generated only when a title is specified; this codepath is hit when you don't specify a name *or* a title. `generateAppName` has a fallback to use the `appPath` to generate a name in the absence of a title; it's an improvement over what was done previously, which generated invalid names if the last segment of the path had e.g. spaces in it.